pkg/ebpf/tracer.go

```diff
@@ -96,18 +96,19 @@ func NewTracer(config *Config) (*Tracer, error) {
 		return nil, fmt.Errorf("could not read bpf module: %s", err)
 	}
 
-	// check if current platform is RHEL or CentOS because it affects what kprobe are we going to enable
-	isRHELOrCentos, err := isRHELOrCentOS()
+	// check if current platform is using old kernel API because it affects what kprobe are we going to enable
+	currKernelVersion, err := CurrentKernelVersion()
 	if err != nil {
-		// if the platform couldn't be determined, treat it as non RHEL case
-		log.Warn("could not detect the platform, will use kprobes from kernel version > 4.1.x")
+		// if the platform couldn't be determined, treat it as new kernel case
+		log.Warn("could not detect the platform, will use kprobes from kernel version >= 4.1.0")
 	}
 
-	if isRHELOrCentos {
-		log.Info("detected platform as RHEL/CentOS, switch to use kprobes from kernel version 3.3.x")
+	isOldKernel := hasOldKernelAPI(currKernelVersion)
+	if isOldKernel {
+		log.Infof("detected old platform %s, switch to use kprobes from kernel version < 4.1.0", kernelCodeToString(currKernelVersion))
 	}
 
-	enableSocketFilter := config.DNSInspection && !isRHELOrCentos
+	enableSocketFilter := config.DNSInspection && !isOldKernel
```

Comment:
This is the DNS check we were talking about, I've made the change for now(since it's disabled so it'll not be affected)